broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ZeebeDbPartitionStep.java

```diff
@@ -54,4 +43,36 @@
   public String getName() {
     return "ZeebeDb";
   }
+
+  private void recoverDb(
+      final PartitionTransitionContext context, final CompletableActorFuture<Void> openFuture) {
+    final ActorFuture<Void> recoverFuture;
+    try {
+      recoverFuture = context.getStateController().recover();
+    } catch (final IOException error) {
+      openFuture.completeExceptionally(
+          String.format(RECOVER_FAILED_ERROR_MSG, context.getPartitionId()), error);
+      return;
+    }
+
+    recoverFuture.onComplete(
+        (ok, error) -> {
+          if (error != null) {
+            openFuture.completeExceptionally(error);
+          } else {
+            try {
+              context.getStateController().verifyDb();
+              final var zeebeDb = context.getStateController().openDb();
```

Comment:
:thought_balloon: question would be then whether verify is really necessary - since we open it here twice

Reply:
`verify` checks if it can open the snapshot. It deletes the runtime if anything fails. I thought of moving it to `openDb`. But I did not want to add this error handling into `openDb`. Previously verification was with in recover. But now copy snapshot is asynchrnous, we cannot verify it within in `recover()`. That's why I split into verify. I am not sure if this is the best way to do it. What do you suggest?

Reply:
IMHO we don't need the line 64. If the opening fails we thrown an exception, which is catched by the block and handled. The deletion is not necessary, since it is anyway not recoverable and will be deleted on the next recover call. :shrug: